apps/root/src/pages/transfer/components/recipient-address/components/contact-list-modal/index.tsx

```diff
@@ -0,0 +1,134 @@
+  const contactList = useStoredContactList();
+  const themeMode = useThemeMode();
+  const intl = useIntl();
+  const [searchValue, setSearchValue] = React.useState('');
+
+  const onOpenAddContactForm = React.useCallback(() => {}, []);
+
+  const noContactsModalContent = React.useMemo(
+    () => (
+      <Grid container justifyContent="center" rowSpacing={6}>
+        <Grid item direction="column" gap={2} textAlign="center" color={colors[themeMode].typography.typo3}>
+          <Typography variant="h3">🫵</Typography>
+          <Typography variant="h5" fontWeight="bold">
+            <FormattedMessage description="noContactsTitle" defaultMessage="Your Contact List Awaits!" />
+          </Typography>
+          <Typography variant="body1">
+            <FormattedMessage
+              description="noContactsDescription"
+              defaultMessage="Looks like you haven't added any contacts yet. Start building your contact list now for easier and faster transactions. Simply click 'Add Contact' to begin."
+            />
+          </Typography>
+        </Grid>
+        <Grid item sx={{ maxWidth: '350px !important' }} xs={12}>
+          <Button variant="contained" onClick={onOpenAddContactForm} fullWidth>
+            <Typography variant="body1" fontWeight="bold">
+              <FormattedMessage description="addContact" defaultMessage="Add Contact" />
+            </Typography>
+          </Button>
+        </Grid>
+      </Grid>
+    ),
+    [onOpenAddContactForm, themeMode]
+  );
+
+  const noContactsOnSearch = React.useMemo(
+    () => (
+      <Grid item direction="column" gap={2} textAlign="center" color={colors[themeMode].typography.typo3}>
+        <Typography variant="h3">🔍</Typography>
+        <Typography variant="body1" fontWeight="bold">
+          <FormattedMessage description="noContactsFound" defaultMessage="No contacts were found" />
+        </Typography>
+      </Grid>
+    ),
+    [themeMode]
+  );
+
+  const filteredContacts = React.useMemo(
+    () =>
+      contactList.filter(
+        (contact) =>
+          contact.address.toLowerCase().includes(searchValue.toLowerCase()) ||
+          contact.label?.toLowerCase().includes(searchValue.toLowerCase())
+      ),
+    [contactList, searchValue]
+  );
+
+  return (
+    <Modal
+      open={shouldShow}
+      onClose={() => setShouldShow(false)}
+      closeOnBackdrop={true}
+      title={<FormattedMessage description="contactList" defaultMessage="Contact list" />}
+      headerButton={
+        contactList.length !== 0 && (
+          <Button variant="outlined" color="primary" onClick={onOpenAddContactForm}>
+            <Typography variant="body1" fontWeight="bold">
+              <FormattedMessage description="addContact" defaultMessage="Add Contact" />
+            </Typography>
+          </Button>
+        )
+      }
+      maxWidth="sm"
+    >
+      <Grid container direction="column" rowSpacing={6}>
+        <Grid item>
```

Comment:
Same comment as before `item` grids should always have their `xs` at least defined